test/e2e/collection/namespace_filtering_test.go

```diff
@@ -0,0 +1,188 @@
+		}
+		rootDir = filepath.Join(filepath.Dir(filename), "..", "..", "..", "/")
+		if elasticsearch, pipelineSecret, err = e2e.DeployAnElasticsearchCluster(rootDir); err != nil {
+			Fail(fmt.Sprintf("Unable to deploy an elastic instance: %v", err))
+		}
+
+		forwarder := &logging.ClusterLogForwarder{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       logging.ClusterLogForwarderKind,
+				APIVersion: logging.GroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "instance",
+			},
+			Spec: logging.ClusterLogForwarderSpec{
+				Inputs: []logging.InputSpec{
+					{
+						Name: "application-logs",
+						Application: &logging.Application{
+							Namespaces: []string{appNamespace1},
+						},
+					},
+				},
+				Outputs: []logging.OutputSpec{
+					{
+						Name: elasticsearch.Name,
+						Secret: &logging.OutputSecretSpec{
+							Name: pipelineSecret.ObjectMeta.Name,
+						},
+						Type: logging.OutputTypeElasticsearch,
+						URL:  fmt.Sprintf("https://%s.%s.svc:9200", elasticsearch.Name, elasticsearch.Namespace),
+					},
+				},
+				Pipelines: []logging.PipelineSpec{
+					{
+						Name:       "test-app",
+						OutputRefs: []string{elasticsearch.Name},
+						InputRefs:  []string{"application-logs"},
+					},
+				},
+			},
+		}
+		if err := e2e.CreateClusterLogForwarder(forwarder); err != nil {
+			Fail(fmt.Sprintf("Unable to create an instance of clusterlogforwarder: %v", err))
+		}
+
+	})
+
+	GetNamespace := func(namespace string) (string, error) {
+		options := metav1.ListOptions{
+			LabelSelector: "component=elasticsearch",
+		}
+		app_index := "app-000001"
+		pods, err := e2e.KubeClient.CoreV1().Pods(constants.OpenshiftNS).List(context.TODO(), options)
+		if err != nil {
+			return "", errors.New("Getting pods error")
+		}
+		if len(pods.Items) == 0 {
+			return "", errors.New("No pods found for elasticsearch")
+		}
+		log.V(3).Info("Pod ", "PodName", pods.Items[0].Name)
+		query := "--query=" + app_index + "/_search?size=10000"
+		stdout, err := e2e.PodExec(constants.OpenshiftNS, pods.Items[0].Name, "elasticsearch", []string{"es_util", query})
+		if err != nil {
+			return "", errors.New("Pods exec error")
+		}
+		// Get total hits
+		var jsonMap map[string]interface{}
+		if err := json.Unmarshal([]byte(stdout), &jsonMap); err != nil {
+			return "", errors.New("json unmarshall error")
+		}
+		total := jsonMap["hits"].(map[string]interface{})["total"]
+		val := int(total.(float64))
+
+		// Parse each document and compare namespace value
+		if val > 0 {
+			docs := jsonMap["hits"].(map[string]interface{})["hits"].([]interface{})
+			for _, document := range docs {
+				namespace_value := document.(map[string]interface{})["_source"].(map[string]interface{})["kubernetes"].(map[string]interface{})["namespace_name"].(string)
+				if namespace_value != namespace {
+					return namespace_value, nil
+				}
+			}
+		}
+		return namespace, nil
+	}
```

Comment:
The function is named `GetNamespace` while taking a namespace name as argument.
why not change the function to return all namespaces from app logs, and then check which namespaces should be present.

Reply:
Logic to retrieve app logs via E2E test is useful outside this test and should be implemented here: https://github.com/openshift/cluster-logging-operator/blob/master/test/framework/e2e/elasticsearch.go#L95  It may be possible to do something clever by refactoring this logic https://github.com/openshift/cluster-logging-operator/blob/master/test/framework/functional/elasticsearch.go#L26  into a place that can be commonly used